src/Servers/Connections.Abstractions/src/UriEndPoint.cs

```diff
@@ -24,5 +24,7 @@ public UriEndPoint(Uri uri)
         /// The <see cref="System.Uri"/> defining the <see cref="EndPoint"/>.
         /// </summary>
         public Uri Uri { get; }
+
+        public override string ToString() => Uri.ToString();
```

Comment:
@davidfowl do you want the encoded or decoded result? ToString is the decoded display value. AbsoluteUri is the encoded useful value.

Reply:
Decoded seems fine.